### ISSUE TITLE
AB3 (java -jar addressbook.jar command not working on Windows Ubuntu terminal)

### ISSUE THREAD

**Issue by jeong-jaeho** [nus-cs2103-AY2324S2/forum#182]
When I run the following command:
![image](https://github.com/nus-cs2103-AY2324S2/forum/assets/109773933/bcb506e9-e073-411c-8d53-1aa87b602466)

I've tried running the command on both Windows terminal and local Ubuntu terminal, but they both throw Exception error message.
My Java version seems to be correctly set as Java 11, but the following error message pops up:

Windows terminal:
![image](https://github.com/nus-cs2103-AY2324S2/forum/assets/109773933/d93002ab-b35b-4f85-968b-83f18e4e57b8)
![image](https://github.com/nus-cs2103-AY2324S2/forum/assets/109773933/7c87db38-b130-49a0-9d47-54ae8c6d1410)

Ubuntu terminal:
![1](https://github.com/nus-cs2103-AY2324S2/forum/assets/109773933/688eae63-73c9-4d5f-9366-3d4a750be414)

**Comment by damithc** [comment]
@jeong-jaeho While waiting for inputs from peers, can you post the output when you run `java -version` in the Windows terminal?

**Comment by Jerome-j** [comment]
Hello @jeong-jaeho I suspect this might be a different issue as #167, #151, #142 and #119. 

It appears that your Ubuntu terminal does not have a GUI? Are you running a GUI such as GNOME etc?

For Windows, I think we need to see your exact Java version using `java --version` command.

**Comment by jeong-jaeho** [comment]
> Hello @jeong-jaeho I suspect this might be a different issue as #167, #151, #142 and #119.
> 
> It appears that your Ubuntu terminal does not have a GUI? Are you running a GUI such as GNOME etc?
> 
> For Windows, I think we need to see your exact Java version using `java --version` command.

I'm only using the Ubuntu CLI without GUI I think, not running Ubuntu on VM but using as Ubuntu 18.04.5 on Windows.
I've updated the post with the java -version command for Windows.

**Comment by Jerome-j** [comment]
Hi @jeong-jaeho your Windows Java is too old, Version 1.8 implies that it is Java 8. https://stackoverflow.com/questions/53642402/why-do-i-see-open-jdk-1-8-instead-of-java-8.

Please upgrade your Java on your Windows machine to Java 11 and above.

Your Linux one is probably working - just Java complaining that there is no GUI to output to. I think you can install a Desktop environment on WSL - but I personally feel it is not worth the effort. If you must use Linux, I recommend that you create a Ubuntu VM on either VirtualBox or VMWare Player. (You can use the Window Hypervisor to create a Ubuntu desktop with the GUI already enabled - but Windows hypervisor has some problems that are not so straightforward that you must know how to deal with.)

To summarise:
1. Upgrade your Windows Java to version 11 (since it is the latest one supported by our course)
2. Create a Ubuntu VM with VirtualBox or VMWare Player will be the easiest.

**Comment by jeong-jaeho** [comment]
> Hi @jeong-jaeho your Windows Java is too old, Version 1.8 implies that it is Java 8. https://stackoverflow.com/questions/53642402/why-do-i-see-open-jdk-1-8-instead-of-java-8.
> 
> Please upgrade your Java on your Windows machine to Java 11 and above.
> 
> Your Linux one is probably working - just Java complaining that there is no GUI to output to. I think you can install a Desktop environment on WSL - but I personally feel it is not worth the effort. If you must use Linux, I recommend that you create a Ubuntu VM on either VirtualBox or VMWare Player. (You can use the Window Hypervisor to create a Ubuntu desktop with the GUI already enabled - but Windows hypervisor has some problems that are not so straightforward that you must know how to deal with.)
> 
> To summarise:
> 
> 1. Upgrade your Windows Java to version 11 (since it is the latest one supported by our course)
> 2. Create a Ubuntu VM with VirtualBox or VMWare Player will be the easiest.

I managed to get the Windows terminal up and running AB3 by updating Java and setting JAVA_HOME to the new update. 
https://docs.oracle.com/cd/E21454_01/html/821-2531/inst_jdk_javahome_t.html

Thanks!
